protocol/communities/manager.go

```diff
@@ -2933,6 +2944,70 @@ func (m *Manager) CheckChannelPermissions(communityID types.HexBytes, chatID str
 	return response, nil
 }
 
+func (m *Manager) checkChannelPermissionsLight(community *Community, communityChat *protobuf.CommunityChat, channelID string) *CheckChannelPermissionsResponse {
+
+	viewOnlyPermissions := community.ChannelTokenPermissionsByType(community.IDString()+channelID, protobuf.CommunityTokenPermission_CAN_VIEW_CHANNEL)
+	viewAndPostPermissions := community.ChannelTokenPermissionsByType(community.IDString()+channelID, protobuf.CommunityTokenPermission_CAN_VIEW_AND_POST_CHANNEL)
+
+	hasViewOnlyPermissions := len(viewOnlyPermissions) > 0
+	hasViewAndPostPermissions := len(viewAndPostPermissions) > 0
+
+	meAsMember := communityChat.Members[common.PubkeyToHex(&m.identity.PublicKey)]
+
+	viewSatisfied := false
+	if !hasViewOnlyPermissions {
+		viewSatisfied = true
+	} else {
+		viewSatisfied = (meAsMember != nil && meAsMember.GetChannelRole() == protobuf.CommunityMember_CHANNEL_ROLE_VIEWER)
+	}
+
+	postSatisfied := false
+	if !hasViewAndPostPermissions {
+		postSatisfied = true
+	} else {
+		postSatisfied = (meAsMember != nil && meAsMember.GetChannelRole() == protobuf.CommunityMember_CHANNEL_ROLE_POSTER)
+	}
+
+	finalViewSatisfied := m.computeViewOnlySatisfied(hasViewOnlyPermissions, hasViewAndPostPermissions, viewSatisfied, postSatisfied)
+	finalPostSatisfied := m.computeViewAndPostSatisfied(hasViewOnlyPermissions, hasViewAndPostPermissions, postSatisfied)
+
+	return &CheckChannelPermissionsResponse{
+		ViewOnlyPermissions: &CheckChannelViewOnlyPermissionsResult{
+			Satisfied:   finalViewSatisfied,
+			Permissions: make(map[string]*PermissionTokenCriteriaResult),
```

Comment:
It's ok to not return criteria results?

Reply:
In light mode yes. Satisfied flag is important. I did not notice any problems with it.